### DISCUSSION TITLE
What's the delay between release and GitHub hosted app deployment?

### DISCUSSION THREAD

**Discussion by TWiStErRob** [renovatebot/renovate#23175] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

I looked and tried to find info about this, but no luck. I would expect this to be documented in one of these places:

* https://docs.renovatebot.com/troubleshooting/#github-hosted-app
* https://docs.renovatebot.com/getting-started/installing-onboarding/#hosted-githubcom-app
* https://docs.renovatebot.com/known-limitations

### What you tried so far.

I have a renovate-config repo, which runs renovate on it so I can see major changes.

I got v36 yesterday https://github.com/TWiStErRob/renovate-config/pull/12

So I went on to update some of the breaking changes in my config files and then got this:
https://github.com/TWiStErRob/net.twisterrob.cinema/issues/121

Apparently I jumped the gun on this: https://docs.renovatebot.com/release-notes-for-major-versions/
![image](https://github.com/renovatebot/renovate/assets/2906988/5d06d183-8daf-4639-b547-fe3fad3cfc6b)

### Relevant debug logs

<details><summary>Logs</summary>

![image](https://github.com/renovatebot/renovate/assets/2906988/b029cfb7-a02e-4a31-8981-cd24ca1a03a9)

</details>

**Comment by rarkins** [comment, marked as answer]
Duplicate of https://github.com/renovatebot/renovate/discussions/22738

**Comment by HonkingGoose** [comment]
Hi @TWiStErRob 👋 

I agree this is confusing! 😄

### Mend-hosted app lags behind source

The `v36` is released on GitHub, so if you're self-hosting you can use it _today_. If you're on the hosted app, you'll have to _wait_ for `@rarkins` to push the update button.

### Why there are no release notes for `v36` in the docs (yet)

I'm updating the **Release notes for major versions of Renovate** docs manually. See PR:

- https://github.com/renovatebot/renovatebot.github.io/pull/295

### Why the docs has a link to `v36` in top right

The version number you see in the top right in the published docs _always_ points to the latest GitHub release on the `renovatebot/renovate` repository. This is a feature of our docs framework, Material for MkDocs:

> If your documentation is related to source code, Material for MkDocs provides the ability to display information to the project's repository as part of the static site, including stars and forks. Furthermore, the [date of last update and creation](https://squidfunk.github.io/mkdocs-material/setup/adding-a-git-repository/#document-dates), as well as [contributors](https://squidfunk.github.io/mkdocs-material/setup/adding-a-git-repository/#document-contributors) can be shown.

Read more in the [Material for MkDocs manual, Adding a Git repository](https://squidfunk.github.io/mkdocs-material/setup/adding-a-git-repository/).

### The Renovate docs lag behind the `renovatebot/renovate` source

The Renovate docs have a Git submodule dependency on the `renovatebot/renovate` repository. We then run Renovate to update that submodule from time to time. [Docs repository `renovate.json` file](https://github.com/renovatebot/renovatebot.github.io/blob/e817678fa0b4b1ea657282f47ffccd1d6cf7566f/renovate.json#L21-L27)

Here's an example of such a submodule update:

- https://github.com/renovatebot/renovatebot.github.io/commit/fc75c9ccd57422d16a212a848f29b88ff182b3a3

### The Renovate docs always follow latest release

Versioning the docs is really hard, so we're only publishing the docs for the _latest_ version. You can find the docs for a older version in the `docs.tgz` file in the corresponding GitHub Release.

**Comment by HonkingGoose** [comment]
@TWiStErRob Quote from the Renovate FAQ: [^1]

> ## When are new Renovate OSS releases added to the hosted Mend Renovate App?
>
> The Renovate maintainers manually update the hosted app. This means the hosted app can lag a few hours to a week behind the open source version.

Can you tell me what keywords you used to search the docs? I can then rewrite the section to include some of those keywords, or do something else so users can find it. 😄 

[^1]: [Renovate Docs, FAQ entry about hosted app](https://docs.renovatebot.com/faq/#when-are-new-renovate-oss-releases-added-to-the-hosted-mend-renovate-app)

**Reply by TWiStErRob** [reply]
I looked at the 4 pages linked in OP, in order I found them. I haven't opened the FAQ, didn't even think there is one because the documentation usually contains relevant info right where it should be.

"Github hosted app" is what I searched for in the Renovate docs.

On Google: https://www.google.com/search?q=renovate+github+app+release+delay and https://www.google.com/search?q=renovate+hosted+app+releases, I think I also tried "Renovate release cadence" and "Renovate release schedule".

One of these actually shows the sentence added, but in a `/blob/` not on the website, so I disregarded it.

Note that https://docs.renovatebot.com/getting-started/running/#mend-renovate-on-premises mentions releasing, I think a mention here would go a long way, because one each of the pages I was Ctrl+F-ing for "release". I think "Renovate OSS" and "When" are not really google-able. For the average user I would guess it's just "Renovate" + "GitHub app", "GitLab app" or "self-hosted".

It's a hard problem because most of the words refer to Renovate features, so they don't match what we're looking for. It also doesn't help that Google is not indexing GitHub discussions too well (I searched for the title of the duplicate discussion and... nothing)

**Reply by HonkingGoose** [reply]
### Hard to find FAQ, move to own entry in sidebar?

> I haven't opened the FAQ, didn't even think there is one because the documentation usually contains relevant info right where it should be.

Yeah, the FAQ is _hidden_ behind the **All Other** dropdown in the sidebar. I remember arguing with @rarkins to move the FAQ to a stand-alone entry. If I remember right, @rarkins definitely wanted it in **All Other** -> **FAQ**. 🙃 

I'm happy to move the FAQ to a stand-alone position in the sidebar, if @rarkins agrees. 😉 

### Alternatives to current FAQ heading

Current heading: _When are new Renovate OSS releases added to the hosted Mend Renovate App?_

Possible alternatives for @rarkins to pick from:

- When is Renovate's GitHub hosted app updated?
- What is Renovate's release schedule?
- Renovate's release schedule for the GitHub hosted app
- Renovate's release cadence
- Something even better than the above

**Reply by rarkins** [reply]
When is Renovate's GitHub hosted app updated?

**Reply by TWiStErRob** [reply]
> How often is Renovate's GitHub hosted app released?

and in the answer use the word "schedule" and "cadence" somehow to cover all grounds.

**Reply by HonkingGoose** [reply]
See PR:

- #23184
